docs/computing/containers/examples.md

```diff
@@ -0,0 +1,129 @@
+# Examples
+
+This section contains examples of building and running containers on Puhti and Mahti.
+
+## Example: Python virtual environment
+
+Next, we provide an example of a container with system Python and virtual environment with Python packages installed using pip.
+We can define the build definition as follows:
+
+```sh title="python-pip.def"
+Bootstrap: docker
+From: docker.io/rockylinux/rockylinux:8.10
+
+%post
+    # Replace the failing commands with always succeeding dummies.
+    cp /usr/bin/true /usr/sbin/useradd
+    cp /usr/bin/true /usr/sbin/groupadd
+
+    # Install Python with system package manager.
+    dnf -y update
+    dnf -y install python3.11 python3.11-pip
+    dnf -y clean all
+
+    # Create a Python virtual environment and install packages using pip.
+    python3.11 -m venv /opt/venv
+    export PATH=/opt/venv/bin:$PATH
+    python3.11 -m pip install --no-cache-dir numpy
+
+%environment
+    export PATH=/opt/venv/bin:$PATH
+```
+
+Now, we can build the container image as follows:
+
+```bash
+apptainer build --fakeroot python-pip.sif python-pip.def
```

Comment:
Doesn't this also need to have `--bind="$TMPDIR:/tmp"` ? Or maybe we should link back to the overview, "see the overview for details on how to run the build command" or similar...